amqp/src/main/scala/akka/stream/alpakka/amqp/model.scala

```diff
@@ -121,15 +121,25 @@ object AmqpConnectionUri {
 }
 
 final case class AmqpConnectionDetails(
-    host: String,
-    port: Int,
+    hostAndPortList: List[(String, Int)],
     credentials: Option[AmqpCredentials] = None,
     virtualHost: Option[String] = None,
-    sslProtocol: Option[String] = None
+    sslProtocol: Option[String] = None,
+    requestedHeartbeat: Option[Int] = None,
+    connectionTimeout: Option[Int] = None,
+    handshakeTimeout: Option[Int] = None,
+    shutdownTimeout: Option[Int] = None,
+    networkRecoveryInterval: Option[Int] = None,
+    automaticRecoveryEnabled: Option[Boolean] = None,
+    topologyRecoveryEnabled: Option[Boolean] = None,
+    exceptionHandler: Option[ExceptionHandler] = None
```

Comment:
Looking at other connectors (amqp and mqtt) the convention is still to use `with***` method name, but also make it a vararg method, like so:

```scala
@annotation.varargs
def withHostAndPorts(hostAndPorts: akka.japi.Pair[String, Int]*) = ...
```